src/Config/ObjectModel/HostOptions.cs

```diff
@@ -1,6 +1,77 @@
+using System.Text.Json.Serialization;
+using Azure.DataApiBuilder.Service.Exceptions;
+
 namespace Azure.DataApiBuilder.Config.ObjectModel;
 
-public record HostOptions(CorsOptions? Cors, AuthenticationOptions? Authentication, HostMode Mode = HostMode.Production);
+public record HostOptions
+{
+    /// <summary>
+    /// Dab engine can at maximum handle 158 MB of data in a single response from a source.
+    /// Json deserialization of a response into a string has a limit of 166,666,666 bytes which when converted to MB is 158 MB.
+    /// ref: enforcing code: https://github.com/dotnet/runtime/blob/main/src/libraries/System.Text.Json/src/System/Text/Json/Writer/JsonWriterHelper.cs#L103
+    /// ref: Json constant: https://github.com/stephentoub/runtime/blob/main/src/libraries/System.Text.Json/src/System/Text/Json/JsonConstants.cs
+    /// </summary>
+    private const int MAX_RESPONSE_LENGTH_DAB_ENGINE_MB = 158;
+
+    /// <summary>
+    /// Dab engine default response length. As of now this is same as max response length.
+    /// </summary>
+    private const int DEFAULT_RESPONSE_LENGTH_DAB_ENGINE_MB = 158;
+
+    [JsonPropertyName("cors")]
+    public CorsOptions? Cors { get; init; }
+
+    [JsonPropertyName("authentication")]
+    public AuthenticationOptions? Authentication { get; init; }
+
+    [JsonPropertyName("mode")]
+    public HostMode Mode { get; init; }
+
+    [JsonPropertyName("max-response-size-mb")]
+    public int? MaxResponseSizeMB { get; init; } = null;
+
+    public HostOptions(CorsOptions? Cors, AuthenticationOptions? Authentication, HostMode Mode = HostMode.Production, int? MaxResponseSizeMB = null)
+    {
+        this.Cors = Cors;
+        this.Authentication = Authentication;
+        this.Mode = Mode;
+        this.MaxResponseSizeMB = MaxResponseSizeMB;
+
+        if (this.MaxResponseSizeMB is not null)
+        {
+            this.MaxResponseSizeMB = this.MaxResponseSizeMB == -1 ? MAX_RESPONSE_LENGTH_DAB_ENGINE_MB : (int)this.MaxResponseSizeMB;
+            if (this.MaxResponseSizeMB < 1 || this.MaxResponseSizeMB > MAX_RESPONSE_LENGTH_DAB_ENGINE_MB)
+            {
+                throw new DataApiBuilderException(
+                    message: $"{nameof(RuntimeConfig.Runtime.Host.MaxResponseSizeMB)} cannot be 0, exceed {MAX_RESPONSE_LENGTH_DAB_ENGINE_MB}MB or be less than -1",
+                    statusCode: HttpStatusCode.ServiceUnavailable,
+                    subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError);
+            }
+
+            UserProvidedMaxResponseSizeMB = true;
+        }
+        else
+        {
+            this.MaxResponseSizeMB = DEFAULT_RESPONSE_LENGTH_DAB_ENGINE_MB;
+        }
+    }
+
+    /// <summary>
+    /// Flag which informs CLI and JSON serializer whether to write MaxResponseSizeMB.
+    /// property and value to the runtime config file.
+    /// When user doesn't provide the MaxResponseSizeMB property/value, which signals DAB to use the default,
```

Comment:
Should clarify that `UserProvidedMaxResponseSizeMB` is true only when a user provides a non-null value. Dab will ignore the property even if the user writes the following in their config:
```json
{
    "max-response-size-mb": null
}
```

Reply:
Ok let me add that in.

Reply:
Added in.